src/index.tsx

```diff
@@ -1,37 +1,46 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import { Root } from './modules/root/components/root';
-import {AddUser, User, UserName} from "./modules/users/user-types";
+import {Root} from './modules/root/components/root';
+import {AddUser, UserName} from "./modules/users/user-types";
+import {compose, createStore} from "redux";
+import {rootReducer} from "./modules/root/root-reducer";
+import {UserActionCreators} from "./modules/users/user-actions";
 
 interface State {
     readonly title: string;
-    readonly users: User[];
 }
 
-let state : State = {
-    title: 'User Management',
-    users: []
+declare global {
+    interface Window {
+        __REDUX_DEVTOOLS_EXTENSION__?: typeof compose
+    }
 }
 
-const addUser: AddUser = (userName: UserName) => {
-    const userToAdd = {
-        ...userName,
-        id: state.users.length
-    };
 
-    state = {
-        ...state,
-        users: [...state.users, userToAdd]
-    };
+const store = createStore(
+    rootReducer,
+    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
+);
 
-    render();
+let state : State = {
+    title: 'User Management'
 }
 
-const render = () => ReactDOM.render(
-    <React.StrictMode>
-        <Root title={state.title} users={state.users} addUser={addUser} />
-    </React.StrictMode>,
-    document.getElementById('root')
-);
+store.subscribe(() => render());
```

Comment:
Jde to zjednodušit na
```suggestion
store.subscribe(render);
```